CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 3.14)
+
+project(pennylane_lightning)
+
+# Read and set pennylane_lightning version
+execute_process(COMMAND "/usr/bin/env" "python"  "-c"
+	"with open('${CMAKE_SOURCE_DIR}/pennylane_lightning/_version.py') as f: print(f.readlines()[-1].split()[-1].strip(\"\\\"'\"), end='')"
+	OUTPUT_VARIABLE VERSION_STRING)
+message(STATUS "Pennylane_lightning version ${VERSION_STRING}")
+set(PROJECT_VERSION ${VERSION_STRING})
+
+set(CMAKE_CXX_STANDARD 11) # At least C++11 is required
+find_package(OpenMP REQUIRED) # find OpenMP
+
+if(NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release)
+endif()
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -W -DVERSION_INFO=${VERSION_STRING}")
+
+
+# Add pybind11
+include(FetchContent)
+FetchContent_Declare(
+	pybind11
+	GIT_REPOSITORY https://github.com/pybind/pybind11.git
+	GIT_TAG        stable 
+)
+FetchContent_MakeAvailable(pybind11)
+
+
+add_library(external_dependency INTERFACE)
+
+
+if (("$ENV{USE_MKL}" OR "${USE_MKL}") AND (DEFINED ENV{MKLROOT}))
+	message(STATUS "Found MKL")
+	target_link_libraries(external_dependency INTERFACE mkl_intel_lp64 mkl_gnu_thread 
+		mkl_core gomp pthread m dl)
+	target_include_directories(external_dependency INTERFACE "$ENV{MKLROOT}/include")
+	target_compile_options(external_dependency INTERFACE "-DMKL=1")
```

Comment:
MKL is always a challenge.

One thing that may cause issues is assuming the user will be using GCC, and hence only require the GNU OpenMP library (what happens if they use icpc and hence want `iomp` ?).

Also, are we always guaranteed to use the LP64 interface here?

Rather than explicitly list the MKL libs, following the [link-line advisor](https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onemkl/link-line-advisor.html), it may be best to opt for linking against the single dynamic library option:
```
 -L${MKLROOT}/lib/intel64 -lmkl_rt -Wl,--no-as-needed -lpthread -lm -ldl
```

and for the headers:

```
 -m64  -I"${MKLROOT}/include"
```

What do you think?

Reply:
Can we just remove these BLAS/LAPACK lines until BLAS/LAPACK is used? Likely Lapack won't be needed anyway, at least it won't be needed for the fundamental operations in a full state vector simulation.

Reply:
I think the same directions as earlier --- rather than creeping the scope, mirroring the existing build process is best here, and we can remove these later across the entire package.

Reply:
As I am not sure what is the best choice, I just removed the MKL part in f0066d3d417c01c0f0f716851f75fc7447dd5ac8 following `setup.py`, as @mlxd suggested.

Reply:
Sure, I think that is fine for now. We will likely rework the BLAS functionality eventually, and MKL support will be part of this.